charts/adguard/templates/primary/route.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.primary.route.enabled -}}
+apiVersion: {{ .Values.primary.route.apiVersion | default "gateway.networking.k8s.io/v1" }}
+kind: {{ .Values.primary.route.kind | default "HTTPRoute" }}
```

Comment:
We don't usually put overwritable defaults in templates, they are usually in values